csharp/extractor/Semmle.Extraction.CSharp/Entities/Expressions/ImplicitCast.cs

```diff
@@ -11,33 +12,74 @@ public Expression Expr
             private set;
         }
 
-        public ImplicitCast(ExpressionNodeInfo info)
+        private ImplicitCast(ExpressionNodeInfo info)
             : base(new ExpressionInfo(info.Context, info.ConvertedType, info.Location, ExprKind.CAST, info.Parent, info.Child, true, info.ExprValue))
         {
             Expr = Factory.Create(new ExpressionNodeInfo(Context, info.Node, this, 0));
         }
 
-        public ImplicitCast(ExpressionNodeInfo info, IMethodSymbol method)
+        private ImplicitCast(ExpressionNodeInfo info, IMethodSymbol method)
             : base(new ExpressionInfo(info.Context, info.ConvertedType, info.Location, ExprKind.OPERATOR_INVOCATION, info.Parent, info.Child, true, info.ExprValue))
         {
             Expr = Factory.Create(info.SetParent(this, 0));
 
+            AddOperatorCall(method);
+        }
+
+        private ImplicitCast(ExpressionInfo info, IMethodSymbol method, object value) : base(info)
+        {
+            Expr = Literal.CreateGenerated(Context, this, 0, method.Parameters[0].Type, value, info.Location);
+
+            AddOperatorCall(method);
+        }
+
+        private void AddOperatorCall(IMethodSymbol method)
+        {
             var target = Method.Create(Context, method);
-            if (target is not null)
-                Context.TrapWriter.Writer.expr_call(this, target);
+            Context.TrapWriter.Writer.expr_call(this, target);
+        }
+
+        private static IMethodSymbol? GetImplicitConversionMethod(ITypeSymbol type, object value) =>
+            type
+                .GetMembers()
+                .Where(m =>
+                    m is IMethodSymbol method &&
+                    method.GetName() == "op_Implicit" &&
+                    method.Parameters.Length == 1 &&
+                    method.Parameters[0].Type.Name == value.GetType().Name
+                )
+                .Cast<IMethodSymbol>()
```

Comment:
You might be able to simplify this `.GetMembers().Where(m => m is IMethodSymbol ...).Cast<IMethodSymbol>()` to `.GetMembers().OfType<IMethodSymbol>().Where(...)`

Reply:
Yes, thank you! I will do that :-)